src/Robo/Common/EnvironmentDetector.php

```diff
@@ -339,7 +339,7 @@ private static function getSubclassResults($functionName) {
         $results[] = call_user_func([$detector, $functionName]);
       }
     }
-    return array_filter($results);
+    return $results;
```

Comment:
The doc block for this function needs to be updated to account for false results being returned.

This significantly changes the behavior of multiple detectors being called, doesn't it? Previously, if any detector returned `true` that detector would 'win', i.e. the parent method would return true. Now, the result is going to vary by method. For instance, what should happen if one detector returns `false` and another returns `true` for `isStageEnv` or `isProdEnv`? I think this behavior needs to be consistent, and it needs to be documented.

Reply:
oh i have some inconsistency here. I had a change of approach last minute.  Which one would be preferred? 

`should happen if one detector returns false and another returns true for isStageEnv or isProdEnv`
This would result in it being overall false. 
If there is only one environment detector returning true - would be true
If there is only one environment detector returning false - would be false (and would not continue to call the parent method)
```
    if ( !empty($results) ) {
      return !in_array(FALSE, $results);
    }
```

`should happen if one detector returns false and another returns true for isStageEnv or isProdEnv`
This would result in it being overall true. 
If there is only one environment detector returning true - would be true
If there is only one environment detector returning false - would be false (and would not continue to call the parent method)
```
    if ( !empty($results) ) {
      return in_array(TRUE, $results);
    }
```